tests/e2e/trace_handlers_test.go

```diff
@@ -0,0 +1,229 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	"github.com/google/go-cmp/cmp"
+)
+
+// an isolated test case for a proposal submission
+var proposalSubmissionSteps = []Step{
+	{submitTextProposalAction{Title: "Proposal 1", Description: "Description 1"}, State{}},
+}
+
+// an isolated test case for a state check involving a proposal
+var proposalInStateSteps = []Step{
+	{
+		Action: submitConsumerRemovalProposalAction{},
+		State: State{
+			ChainID("provi"): ChainState{
+				Proposals: &map[uint]Proposal{
+					1: ConsumerRemovalProposal{
+						Deposit:  10000001,
+						Chain:    ChainID("foo"),
+						StopTime: 0,
+						Status:   "PROPOSAL_STATUS_VOTING_PERIOD",
+					},
+				},
+			},
+		},
+	},
+}
+
+// Checks that writing, then parsing a trace results in the same trace.
+func TestWriterThenParser(t *testing.T) {
+	tests := map[string]struct {
+		trace []Step
+	}{
+		"proposalSubmission":    {proposalSubmissionSteps},
+		"proposalInState":       {proposalInStateSteps},
+		"start_provider_chain":  {stepStartProviderChain()},
```

Comment:
Nit, make `stepStartProviderChain` a var instead of func to be consistent with all the other fields

Reply:
I think this should stay as-is - this is not a one-off function, it's called in many places.